Fix flaky admission controller unit test

I've seen multiple instances of this on unrelated PRs: 

?   	k8s.io/autoscaler/vertical-pod-autoscaler/common	[no test files]
?   	k8s.io/autoscaler/vertical-pod-autoscaler/pkg/admission-controller	[no test files]
--- FAIL: TestGetPatchesForResourceRequest_TwoReplacementResources (0.00s)
    server_test.go:396: 
        	Error Trace:	server_test.go:396
        	Error:      	Should be true
        	Test:       	TestGetPatchesForResourceRequest_TwoReplacementResources
FAIL
FAIL	k8s.io/autoscaler/vertical-pod-autoscaler/pkg/admission-controller/logic	0.097s

In #2905 I just saw a failure with new [logs](https://travis-ci.org/kubernetes/autoscaler/jobs/660079355?utm_medium=notification&utm_source=github_status):

```
--- FAIL: TestGetPatchesForResourceRequest_TwoReplacementResources (0.00s)
    server_test.go:136: 
        	Error Trace:	server_test.go:136
        	            				server_test.go:410
        	Error:      	Should be true
        	Test:       	TestGetPatchesForResourceRequest_TwoReplacementResources
        	Messages:   	got {Op:add Path:/metadata/annotations/vpaUpdates Value:Pod resources updated by name: container 0: unobtanium request, cpu request}, want: {Op:add Path:/metadata/annotations/vpaUpdates Value:Pod resources updated by name: container 0: cpu request, unobtanium request}

```

--- a/vertical-pod-autoscaler/pkg/admission-controller/logic/server_test.go
+++ b/vertical-pod-autoscaler/pkg/admission-controller/logic/server_test.go
@@ -407,7 +407,9 @@ func TestGetPatchesForResourceRequest_TwoReplacementResources(t *testing.T) {
 		assertPatchOneOf(t, patches[1], []patchRecord{cpuUpdate, unobtaniumUpdate})
 		assert.False(t, eqPatch(patches[0], patches[1]))
 		assertEqPatch(t, patches[2], getAddEmptyAnnotationsPatch())
-		assertEqPatch(t, patches[3], addAnnotationRequest([][]string{{cpu, unobtanium}}, request))
+		cpuFirstUnobtaniumSecond := addAnnotationRequest([][]string{{cpu, unobtanium}}, request)
+		unobtaniumFirstCpuSecond := addAnnotationRequest([][]string{{unobtanium, cpu}}, request)
+		assertPatchOneOf(t, patches[3], []patchRecord{cpuFirstUnobtaniumSecond, unobtaniumFirstCpuSecond})
 		assertEqPatch(t, patches[4], addVpaObservedContainersPatch([]string{}))
 	}
 }

